Mathlib/Algebra/SkewMonoidAlgebra/Basic.lean

```diff
@@ -0,0 +1,1465 @@
+  Function.Injective.isCancelAdd toFinsupp toFinsupp_injective toFinsupp_add
+
+instance coeFun : CoeFun (SkewMonoidAlgebra k G) fun _ => G → k := ⟨fun s ↦ s.toFinsupp⟩
+
+@[simp, norm_cast]
+theorem coe_smul [SMulZeroClass R k] (b : R) (v : SkewMonoidAlgebra k G) :
+    ⇑(b • v) = b • ⇑v :=
+  rfl
+
+section Support
+
+/-- For `f : SkewMonoidAlgebra k G`, `f.support` is the set of all `a ∈ G` such that
+`f a ≠ 0`. -/
+def support : SkewMonoidAlgebra k G → Finset G
+  | ⟨p⟩ => p.support
+
+@[simp]
+theorem support_ofFinsupp (p) : support (⟨p⟩ : SkewMonoidAlgebra k G) = p.support := by
+  rw [support]
+
+theorem support_toFinsupp (p : SkewMonoidAlgebra k G) : p.toFinsupp.support = p.support := by
+  rw [support]
+
+@[simp]
+theorem support_zero : (0 : SkewMonoidAlgebra k G).support = ∅ := rfl
+
+@[simp]
+theorem support_eq_empty {p} : p.support = ∅ ↔ (p : SkewMonoidAlgebra k G) = 0 := by
+  rcases p with ⟨⟩
+  simp only [support, Finsupp.support_eq_empty, ofFinsupp_eq_zero]
+
+@[simp]
+theorem mem_support_iff {f : SkewMonoidAlgebra k G} {a : G} : a ∈ f.support ↔ f a ≠ 0 := by
+  rcases f with ⟨⟩
+  simp only [support_ofFinsupp, Finsupp.mem_support_iff, ne_eq, implies_true]
+
+end Support
+
+section Single
+
+/-- `single a b` is the finitely supported function with value `b` at `a` and zero otherwise. -/
+def single (a : G) (b : k) : SkewMonoidAlgebra k G := ⟨Finsupp.single a b⟩
+
+@[simp]
+theorem toFinsupp_single (a : G) (b : k) : (single a b).toFinsupp = Finsupp.single a b := rfl
+
+@[simp]
+theorem ofFinsupp_single (a : G) (b : k) : ⟨Finsupp.single a b⟩ = single a b := rfl
+
+theorem single_zero_right (a : G) : single a (0 : k) = 0 := by
+  ext; simp only [Finsupp.single_zero, single, Finsupp.coe_zero, Pi.zero_apply, toFinsupp_zero]
+
+@[simp]
+theorem single_add (a : G) (b₁ b₂ : k) : single a (b₁ + b₂) = single a b₁ + single a b₂ := by
+  ext; simp only [Finsupp.single_add, single, Finsupp.coe_add, Pi.add_apply, toFinsupp_add]
+
+theorem single_zero (a : G) : (single a 0 : SkewMonoidAlgebra k G) = 0 := by
+  rw [ofFinsupp_eq_zero, single, Finsupp.single_zero]
+
+theorem single_apply {a a' : G} {b : k} [Decidable (a = a')] :
+    single a b a' = if a = a' then b else 0 :=
+  Finsupp.single_apply
+
+theorem single_eq_zero {a : G} {b : k} : single a b = 0 ↔ b = 0 := by
+  simp only [ofFinsupp_eq_zero, single, Finsupp.single_eq_zero]
+
+theorem toFinsupp_single' : single = (fun (a : G) (b : k) ↦ ofFinsupp (Finsupp.single a b)) := rfl
+
+/-- Group isomorphism between `SkewMonoidAlgebra k G` and `G →₀ k`. This is an
+implementation detail, but it can be useful to transfer results from `Finsupp`
+to `SkewMonoidAlgebra`. -/
+@[simps apply symm_apply]
+def toFinsuppAddEquiv : SkewMonoidAlgebra k G ≃+ (G →₀ k) where
+  toFun        := toFinsupp
+  invFun       := ofFinsupp
+  left_inv     := fun ⟨_p⟩ => rfl
+  right_inv _p := rfl
+  map_add'     := toFinsupp_add
+
+theorem smul_single {S} [SMulZeroClass S k] (s : S) (a : G) (b : k) :
```

Comment:
Is it worth adding that `toFinsuppAddEquiv (single a b) = single a b`?

Reply:
This already exists thanks to the `@[apply]` tag.